### DISCUSSION TITLE
Redirect with 301 - subdomain

### DISCUSSION THREAD

**Discussion by gutovysk** [community/community#22793] [Other Feature Feedback, Questions, & Ideas]
<p>Hi everyone,</p>
<p>I have a blog on blogger with domain www.mysite.com.br and mysite.com.br.</p>
<p>Also, I have a site on github pages for static files with domain  <strong>pages.mysite.com.br</strong>.</p>
<p>Now, i want to change the subdomain like this permanently:<br>
blog (blogger):   <strong>blog.mysite.com.br</strong><br>
pages (github pages):  <strong>www.mysite.com.br</strong></p>
<p>I searched for how can I change with permanent redirect 301 but, I think, this is not possible on github pages, because we don’t have access to .htaccess. That’s right?</p>
<p>I don’t know how to redirect subdomain on github pages with permanent redirect:<br>
<strong>pages</strong>.mysite.com.br  <strong>to</strong>   <strong>www</strong>.mysite.com.br    <strong>and</strong>   mysite.com.br</p>
<p>So, how can I resolve this without loosing my datas on search engines and doesn’t break these links???</p>
<p>Is there a way to do this?</p>
<p>Tks a lot,<br>
Carlos Correia</p>

**Comment by otter-computer** [comment]
<p>Hi <a class="mention" href="/gutovysk">@gutovysk</a>, welcome to the GitHub Support Community!</p>
<p>There isn’t a way to do this directly on GitHub, no. GitHub doesn’t give you the ability to modify webserver settings using an <code>.htaccess</code> file.</p>
<p>However, you may be able to achieve this using your domain provider/DNS host. Certain domain providers, such as Google Domains, allow you to set a 301 redirect at the domain level. I’d recommend checking with your domain or DNS provider if this is possible.</p>

**Comment by gutovysk** [comment]
<p>Tks Thomas!</p>
<p>But, do you know how to do a redirect in DNS, what instruction is? Is using a TXT, CNAME, or other instruction?</p>

**Comment by otter-computer** [comment, marked as answer]
<p>It depends based on your domain provider, but usually there’s an option in their settings panel if they’re able to support it. I’d recommend either reading your provider’s support documentation or asking their support team directly.</p>
<p>You <em>can</em> set up a <code>CNAME</code> record to alias your old domain to your new one, but this could result in poorer SEO, so a redirect is probably preferable if you’re able to set it up this way.</p>

**Comment by wgailey** [comment]
After much trouble I found a way to do 301 redirects without even touching github stuff. Cloudflare (the best dns provider I've ever had and I've had a few) offers 301 redirects (that even Google accepts) through page rules.
